dynawo/sources/Solvers/SolverIDA/DYNSolverIDA.cpp

```diff
@@ -125,15 +126,32 @@ SolverIDA::SolverIDA() {
   flagInit_ = false;
 
   IDAMem_ = NULL;
+  M_ = NULL;
+  LS_ = NULL;
   // KINSOL solver Init
   //-----------------------
   solverKIN_.reset(new SolverKIN());
 
   lastRowVals_ = NULL;
+  order_ = 0;
+  initStep_ = 0.;
+  minStep_ = 0.;
+  maxStep_ = 0.;
+  absAccuracy_ = 0.;
+  relAccuracy_ = 0.;
+  flagInit_ = false;
 }
 
 void
 SolverIDA::clean() {
+  if (M_ != NULL) {
+    SUNMatDestroy(M_);
+    M_ = NULL;
```

Comment:
I get the impression that the SUNMatDestroy makes the M_ = NULL for you and that you don't have to do it again: free(A); A = NULL;

Reply:
I would prefer not relying on a side effect of a third party library to avoid having dangling pointers

Reply:
Ok for me. It makes sense